subsys/bluetooth/controller/ll_sw/ull_scan_aux.c

```diff
@@ -1422,6 +1430,11 @@ static void flush(void *param)
 	scan = HDR_LLL2ULL(lll);
 	scan = ull_scan_is_valid_get(scan);
 	if (!IS_ENABLED(CONFIG_BT_CTLR_SYNC_PERIODIC) || scan) {
+		/* Remove auxiliary context association with scan context */
+		if (lll->lll_aux == &aux->lll) {
+			lll->lll_aux = NULL;
+		}
+
```

Comment:
This is also add missing code already in the other design variant here:
https://github.com/zephyrproject-rtos/zephyr/blob/60b89ec06009717e6f9030778cb8e0b409b20cc4/subsys/bluetooth/controller/ll_sw/ull_scan_aux.c#L2685-L2687